### ISSUE TITLE
Gracefully handle re-auth

### ISSUE THREAD

**Issue by BenWiederhake** [majn/telegram-purple#226]
Apparently, there is a common thing about all those errors of this type:

```
(13:40:12) prpl-telegram: error for query 'get difference' #6241894945970085936: #401 AUTH_KEY_UNREGISTERED
(13:40:12) prpl-telegram: login problem: error #134 (RPC_CALL_FAIL 401: AUTH_KEY_UNREGISTERED)
<crash>
```

This might be related to users pressing the "close all open sessions" button in other apps. I guess that tgp can't yet gracefully handle re-authentication or having a "wrong" token.

**Comment by majn** [comment]
I created a patch for this in dev.1.2.5: https://github.com/majn/telegram-purple/commit/cdb3e8409581b18e379a0a58089fa10446ad8a45

**Comment by BenWiederhake** [comment]
Have you tested that? Just looking at the code and wondering what it does when the option `reset-authorization` is set to false:

    purple_connection_set_state (conn->gc, PURPLE_CONNECTING);
    tgl_login (TLS);

So if the user sets this option fo `FALSE`, then tgp tells libpurple that it's reconnecting, *still* attempts to login, but doesn't reset authorization. I believe this isn't useful.

Is that option meaningful at all? I suggest getting rid of the switch and *always* attempting to re-auth when necessary.

**Comment by majn** [comment]
Yeah, I've tested it and it worked for me.

This isn't an option, I just use it to store an account state, its not visible to the user. When the connection fails due to a dropped session (SESSION_REVOKED, AUTH_KEY_UNREGISTERED when logging into a revoked session), that setting will remember it and attempt a new login on the next connection attempt.

Note that this works like telegram-cli, which tells the user to delete ~/.telegram-cli/auth or to pass the -q flag, since the session revokation happens on the next start of tgl. The difference between tgp and tg  is that in my implementation it happens automatically on the next restart (I suspect that the reason why its not automated in telegram-cli is probably cause it doesn't have the ability to restart itself).

**Comment by majn** [comment]
Implemented in 1.2.5
